{"uid":"a79d5b6d3f22eb2c","name":"test_dynamic_steps","fullName":"tests.test_steps#test_dynamic_steps","historyId":"7c7c5e5f20fe2663d33f122f5ce2ca5a","time":{"start":1697444620101,"stop":1697444620102,"duration":1},"status":"broken","statusMessage":"ImportError: cannot import name 'ChromeType' from 'webdriver_manager.core.utils' (/opt/hostedtoolcache/Python/3.11.5/x64/lib/python3.11/site-packages/webdriver_manager/core/utils.py)","statusTrace":"browser_management = None\n\n    def test_dynamic_steps(browser_management):\n        with allure.step('Open main page'):\n>           browser.open('/')\n\ntests/test_steps.py:10: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n/opt/hostedtoolcache/Python/3.11.5/x64/lib/python3.11/site-packages/selene/core/entity.py:1154: in open\n    self.config._executor.get_url(relative_or_absolute_url)\n/opt/hostedtoolcache/Python/3.11.5/x64/lib/python3.11/site-packages/selene/core/configuration.py:303: in get_url\n    self.config._driver_get_url_strategy(self.config)(url)\n/opt/hostedtoolcache/Python/3.11.5/x64/lib/python3.11/site-packages/selene/core/configuration.py:218: in get\n    driver = config.driver\n/opt/hostedtoolcache/Python/3.11.5/x64/lib/python3.11/site-packages/selene/core/configuration.py:345: in __get__\n    driver = config.build_driver_strategy(config)\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\nconfig = <selene.core.configuration.Config object at 0x7ff7e625de90>\n\n    def _build_local_driver_by_name_or_remote_by_url_and_options(\n        config: Config,\n    ) -> WebDriver:\n        from selenium.webdriver import (\n            ChromeOptions,\n            EdgeOptions,\n            Chrome,\n            Firefox,\n            Edge,\n        )\n    \n        from selenium.webdriver.chrome.service import Service as ChromeService\n        from selenium.webdriver.firefox.service import Service as FirefoxService\n        from selenium.webdriver.edge.service import Service as EdgeService  # type: ignore\n    \n        from webdriver_manager.chrome import ChromeDriverManager  # type: ignore\n        from webdriver_manager.firefox import GeckoDriverManager  # type: ignore\n        from webdriver_manager.microsoft import EdgeChromiumDriverManager  # type: ignore\n    \n>       from webdriver_manager.core.utils import ChromeType  # type: ignore\nE       ImportError: cannot import name 'ChromeType' from 'webdriver_manager.core.utils' (/opt/hostedtoolcache/Python/3.11.5/x64/lib/python3.11/site-packages/webdriver_manager/core/utils.py)\n\n/opt/hostedtoolcache/Python/3.11.5/x64/lib/python3.11/site-packages/selene/core/configuration.py:69: ImportError","flaky":false,"newFailed":false,"newBroken":true,"newPassed":false,"retriesCount":0,"retriesStatusChange":false,"beforeStages":[{"name":"browser_management","time":{"start":1697444620101,"stop":1697444620101,"duration":0},"status":"passed","steps":[],"attachments":[],"parameters":[],"shouldDisplayMessage":false,"stepsCount":0,"attachmentsCount":0,"hasContent":false}],"testStage":{"status":"broken","statusMessage":"ImportError: cannot import name 'ChromeType' from 'webdriver_manager.core.utils' (/opt/hostedtoolcache/Python/3.11.5/x64/lib/python3.11/site-packages/webdriver_manager/core/utils.py)","statusTrace":"browser_management = None\n\n    def test_dynamic_steps(browser_management):\n        with allure.step('Open main page'):\n>           browser.open('/')\n\ntests/test_steps.py:10: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n/opt/hostedtoolcache/Python/3.11.5/x64/lib/python3.11/site-packages/selene/core/entity.py:1154: in open\n    self.config._executor.get_url(relative_or_absolute_url)\n/opt/hostedtoolcache/Python/3.11.5/x64/lib/python3.11/site-packages/selene/core/configuration.py:303: in get_url\n    self.config._driver_get_url_strategy(self.config)(url)\n/opt/hostedtoolcache/Python/3.11.5/x64/lib/python3.11/site-packages/selene/core/configuration.py:218: in get\n    driver = config.driver\n/opt/hostedtoolcache/Python/3.11.5/x64/lib/python3.11/site-packages/selene/core/configuration.py:345: in __get__\n    driver = config.build_driver_strategy(config)\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\nconfig = <selene.core.configuration.Config object at 0x7ff7e625de90>\n\n    def _build_local_driver_by_name_or_remote_by_url_and_options(\n        config: Config,\n    ) -> WebDriver:\n        from selenium.webdriver import (\n            ChromeOptions,\n            EdgeOptions,\n            Chrome,\n            Firefox,\n            Edge,\n        )\n    \n        from selenium.webdriver.chrome.service import Service as ChromeService\n        from selenium.webdriver.firefox.service import Service as FirefoxService\n        from selenium.webdriver.edge.service import Service as EdgeService  # type: ignore\n    \n        from webdriver_manager.chrome import ChromeDriverManager  # type: ignore\n        from webdriver_manager.firefox import GeckoDriverManager  # type: ignore\n        from webdriver_manager.microsoft import EdgeChromiumDriverManager  # type: ignore\n    \n>       from webdriver_manager.core.utils import ChromeType  # type: ignore\nE       ImportError: cannot import name 'ChromeType' from 'webdriver_manager.core.utils' (/opt/hostedtoolcache/Python/3.11.5/x64/lib/python3.11/site-packages/webdriver_manager/core/utils.py)\n\n/opt/hostedtoolcache/Python/3.11.5/x64/lib/python3.11/site-packages/selene/core/configuration.py:69: ImportError","steps":[{"name":"Open main page","time":{"start":1697444620101,"stop":1697444620101,"duration":0},"status":"broken","statusMessage":"ImportError: cannot import name 'ChromeType' from 'webdriver_manager.core.utils' (/opt/hostedtoolcache/Python/3.11.5/x64/lib/python3.11/site-packages/webdriver_manager/core/utils.py)\n","statusTrace":"  File \"/home/runner/work/qa_guru_python_7/qa_guru_python_7/tests/test_steps.py\", line 10, in test_dynamic_steps\n    browser.open('/')\n  File \"/opt/hostedtoolcache/Python/3.11.5/x64/lib/python3.11/site-packages/selene/core/entity.py\", line 1154, in open\n    self.config._executor.get_url(relative_or_absolute_url)\n  File \"/opt/hostedtoolcache/Python/3.11.5/x64/lib/python3.11/site-packages/selene/core/configuration.py\", line 303, in get_url\n    self.config._driver_get_url_strategy(self.config)(url)\n  File \"/opt/hostedtoolcache/Python/3.11.5/x64/lib/python3.11/site-packages/selene/core/configuration.py\", line 218, in get\n    driver = config.driver\n             ^^^^^^^^^^^^^\n  File \"/opt/hostedtoolcache/Python/3.11.5/x64/lib/python3.11/site-packages/selene/core/configuration.py\", line 345, in __get__\n    driver = config.build_driver_strategy(config)\n             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/opt/hostedtoolcache/Python/3.11.5/x64/lib/python3.11/site-packages/selene/core/configuration.py\", line 69, in _build_local_driver_by_name_or_remote_by_url_and_options\n    from webdriver_manager.core.utils import ChromeType  # type: ignore\n    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","steps":[],"attachments":[],"parameters":[],"shouldDisplayMessage":true,"stepsCount":0,"attachmentsCount":0,"hasContent":true}],"attachments":[],"parameters":[],"shouldDisplayMessage":true,"stepsCount":1,"attachmentsCount":0,"hasContent":true},"afterStages":[{"name":"browser_management::0","time":{"start":1697444620131,"stop":1697444620131,"duration":0},"status":"broken","statusMessage":"ImportError: cannot import name 'ChromeType' from 'webdriver_manager.core.utils' (/opt/hostedtoolcache/Python/3.11.5/x64/lib/python3.11/site-packages/webdriver_manager/core/utils.py)\n","statusTrace":"  File \"/opt/hostedtoolcache/Python/3.11.5/x64/lib/python3.11/site-packages/allure_commons/_allure.py\", line 221, in __call__\n    return self._fixture_function(*args, **kwargs)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/opt/hostedtoolcache/Python/3.11.5/x64/lib/python3.11/site-packages/_pytest/fixtures.py\", line 911, in _teardown_yield_fixture\n    next(it)\n  File \"/home/runner/work/qa_guru_python_7/qa_guru_python_7/tests/conftest.py\", line 21, in browser_management\n    browser.quit()\n  File \"/opt/hostedtoolcache/Python/3.11.5/x64/lib/python3.11/site-packages/selene/core/entity.py\", line 1204, in quit\n    self.driver.quit()\n    ^^^^^^^^^^^\n  File \"/opt/hostedtoolcache/Python/3.11.5/x64/lib/python3.11/site-packages/selene/core/entity.py\", line 1099, in driver\n    return self.config.driver\n           ^^^^^^^^^^^^^^^^^^\n  File \"/opt/hostedtoolcache/Python/3.11.5/x64/lib/python3.11/site-packages/selene/core/configuration.py\", line 345, in __get__\n    driver = config.build_driver_strategy(config)\n             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/opt/hostedtoolcache/Python/3.11.5/x64/lib/python3.11/site-packages/selene/core/configuration.py\", line 69, in _build_local_driver_by_name_or_remote_by_url_and_options\n    from webdriver_manager.core.utils import ChromeType  # type: ignore\n    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","steps":[],"attachments":[],"parameters":[],"shouldDisplayMessage":true,"stepsCount":0,"attachmentsCount":0,"hasContent":true}],"labels":[{"name":"parentSuite","value":"tests"},{"name":"suite","value":"test_steps"},{"name":"host","value":"fv-az398-616"},{"name":"thread","value":"2190-MainThread"},{"name":"framework","value":"pytest"},{"name":"language","value":"cpython3"},{"name":"package","value":"tests.test_steps"},{"name":"resultFormat","value":"allure2"}],"parameters":[],"links":[],"hidden":false,"retry":false,"extra":{"severity":"normal","retries":[],"categories":[{"name":"Test defects","matchedStatuses":[],"flaky":false}],"history":{"statistic":{"failed":0,"broken":3,"skipped":0,"passed":2,"unknown":0,"total":5},"items":[{"uid":"651e0704323e7ff1","reportUrl":"https://vapolonov.github.io/qa_guru_python_7/9//#testresult/651e0704323e7ff1","status":"passed","time":{"start":1688982507733,"stop":1688982513518,"duration":5785}},{"uid":"bca2a94cf205c615","reportUrl":"https://vapolonov.github.io/qa_guru_python_7/8//#testresult/bca2a94cf205c615","status":"passed","time":{"start":1688981960291,"stop":1688981966150,"duration":5859}},{"uid":"75d0c1391bbc46cb","reportUrl":"https://vapolonov.github.io/qa_guru_python_7/7//#testresult/75d0c1391bbc46cb","status":"broken","statusDetails":"selenium.common.exceptions.WebDriverException: Message: unknown error: Chrome failed to start: exited abnormally.\n  (unknown error: DevToolsActivePort file doesn't exist)\n  (The process started from chrome location /usr/bin/google-chrome is no longer running, so ChromeDriver is assuming that Chrome has crashed.)\nStacktrace:\n#0 0x55f38a43e4e3 <unknown>\n#1 0x55f38a16dc76 <unknown>\n#2 0x55f38a196d78 <unknown>\n#3 0x55f38a193029 <unknown>\n#4 0x55f38a1d1ccc <unknown>\n#5 0x55f38a1d147f <unknown>\n#6 0x55f38a1c8de3 <unknown>\n#7 0x55f38a19e2dd <unknown>\n#8 0x55f38a19f34e <unknown>\n#9 0x55f38a3fe3e4 <unknown>\n#10 0x55f38a4023d7 <unknown>\n#11 0x55f38a40cb20 <unknown>\n#12 0x55f38a403023 <unknown>\n#13 0x55f38a3d11aa <unknown>\n#14 0x55f38a4276b8 <unknown>\n#15 0x55f38a427847 <unknown>\n#16 0x55f38a437243 <unknown>\n#17 0x7f215f4c0b43 <unknown>","time":{"start":1688980174068,"stop":1688980174713,"duration":645}},{"uid":"380fb46da76d2bf8","reportUrl":"https://vapolonov.github.io/qa_guru_python_7/5//#testresult/380fb46da76d2bf8","status":"broken","statusDetails":"selenium.common.exceptions.WebDriverException: Message: unknown error: Chrome failed to start: exited abnormally.\n  (unknown error: DevToolsActivePort file doesn't exist)\n  (The process started from chrome location /usr/bin/google-chrome is no longer running, so ChromeDriver is assuming that Chrome has crashed.)\nStacktrace:\n#0 0x5565bba2d4e3 <unknown>\n#1 0x5565bb75cc76 <unknown>\n#2 0x5565bb785d78 <unknown>\n#3 0x5565bb782029 <unknown>\n#4 0x5565bb7c0ccc <unknown>\n#5 0x5565bb7c047f <unknown>\n#6 0x5565bb7b7de3 <unknown>\n#7 0x5565bb78d2dd <unknown>\n#8 0x5565bb78e34e <unknown>\n#9 0x5565bb9ed3e4 <unknown>\n#10 0x5565bb9f13d7 <unknown>\n#11 0x5565bb9fbb20 <unknown>\n#12 0x5565bb9f2023 <unknown>\n#13 0x5565bb9c01aa <unknown>\n#14 0x5565bba166b8 <unknown>\n#15 0x5565bba16847 <unknown>\n#16 0x5565bba26243 <unknown>\n#17 0x7f3ac113ab43 <unknown>","time":{"start":1688979000358,"stop":1688979001004,"duration":646}}]},"tags":[]},"source":"a79d5b6d3f22eb2c.json","parameterValues":[]}